### PR TITLE
Upgrade to React 16

### PR DESCRIPTION
**Pull request by djbeaumont**
* Supports React 16 without dropping support for React 14 (Removes `isMounted()` call, uses es2015 class)
* Fixes showing nice duration animation without causing prop type warnings
* Uses only enzyme for tests.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+  "version": "4.0.0",
   "description": "Component to generate a simple loading bar (wrapped in a panel)",
   "main": "dist/ProgressBar.js",
   "scripts": {
     "prepublish": "npm run build",
     "build": "mkdir -p dist && babel src/ProgressBar.jsx -o dist/ProgressBar.js",
     "prestart": "ulimit -n 9999",
     "start": "watchify -t babelify doc/example.js -o doc/example-built.js -v",
-    "test": "mocha --compilers js:babel-register -r test/setup test/test*",
+    "test": "mocha --require babel-register -r test/setup test/test*",
```
Comment (samsmarthx) [inline package.json]: Do we need the exit flag as part of moving to mocha 4?

File: test/testProgressBar.jsx [hunk 2]
```diff
@@ -18,17 +16,13 @@ const defaultProps = {
   duration: 1,
   title: null,
   subtitle: null,
-  type: 'info'
+  type: 'success'
```
Comment (samsmarthx) [inline test/testProgressBar.jsx]: Does this change its colour?

File: package.json (hunk 1 above)
Comment (djbeaumont) [reply package.json]: I think if the build was successful the answer is 'no'. Ideally that flag's not needed and is a bit of a warning that your tests are kicking off async actions that they don't wait for.

File: test/testProgressBar.jsx (hunk 2 above)
Comment (samsmarthx) [reply test/testProgressBar.jsx]: This is a test, ignore me...

**Comment by samsmarthx** [comment]
Looks good, I'm tempted to say this might be one of the components that we should look at where its being used and whether it needs its own repo 👍 :shipit:

**Pull request merged**